src/db_state.rs

```diff
@@ -384,10 +400,12 @@ impl DbState {
             new_l0.push_back(sst.clone());
         }
         let compacted = compactor_state.compacted.clone();
+        let checkpoints = compactor_state.checkpoints.clone();
```

Comment:
Note this appears to be a bug. Checkpoints are dropped from the new manifest. I can fix this separately if that is preferred.

Reply:
I think it's ok - creating a checkpoint forces a manifest write, and the compactor will be forced to load the new manifest (and the contained checkpoints) before updating it.

Reply:
Fixed in #435 and #437.